t/readlink.t

```diff
@@ -49,13 +49,21 @@ $! = 0;
 my $got = 'abc';
 like( warning { $got = CORE::readlink(undef) }, qr/^Use of uninitialized value in readlink at /, "Got expected warning for passing no value to readlink" );
 is( $got,   undef,  "readlink without args is undef." );
-is( $! + 0, ENOENT, '$! is ENOENT for a readlink(undef)' );
+if ( $^O eq 'freebsd' ) {
+    is( $! + 0, EINVAL, '$! is EINVAL for a readlink(undef)' );
```

Comment:
The point of this test os to validate that pure perl behaves the same as us. How are we making T:MF behave like this?

Reply:
I have updated the code to throw `EINVAL` instead of `ENOENT` for FreeBSD. The frustrating bit is that the manual doesn't say `undef` will return `EINVAL` so it's hard to know what the cases are exactly or whether it's a change in some versions of FreeBSD, etc. I guess we'll see.